src/components/Onboarding/transaction-logic.js

```diff
@@ -0,0 +1,238 @@
+import { bigNum } from '@lib/bigNumber'
+import { getNetwork } from '@/networks'
+import { ZERO_ADDR } from '@/constants'
+import { getContract } from '@hooks/useContract'
+import { uploadToPinata } from '@/services/pinata'
+import { encodeFunctionData } from '@utils/web3-utils'
+import { BYOT_TYPE, NATIVE_TYPE } from '@components/Onboarding/constants'
+
+import tokenAbi from '@abis/erc20.json'
+import templateAbi from '@abis/gardensTemplate.json'
+
+const C_V_ONE_HUNDRED_PERCENT = 1e7
+const ISSUANCE_ONE_HUNDRED_PERCENT = 1e10
+const ONE_HUNDRED_PCT = 1e18
+
+export function createTokenApproveTxs({ garden, liquidity, tokens }) {
+  const txs = []
+  const network = getNetwork()
+  const templateAddress = network.template
+  const honeyTokenAddress = network.honeyToken
+
+  const adjustedHoneyLiquidity = bigNum(
+    liquidity.honeyTokenLiquidity
+  ).toString()
+
+  txs.push(
+    createTokenTx(honeyTokenAddress, 'approve', [
+      templateAddress,
+      adjustedHoneyLiquidity,
+    ])
+  )
+
+  if (garden.type === BYOT_TYPE) {
+    const adjustedTokenLiquidity = bigNum(liquidity.tokenLiquidity).toString()
+
+    txs.push(
+      createTokenTx(tokens.address, 'approve', [
+        templateAddress,
+        adjustedTokenLiquidity,
+      ])
+    )
+  }
+
+  return txs
+}
+
+export function createGardenTxOne({
+  garden,
+  issuance,
+  liquidity,
+  tokens,
+  voting,
+}) {
+  let existingToken, commonPool, gardenTokenLiquidity, existingTokenLiquidity
+
+  // New token
+  if (garden.type === NATIVE_TYPE) {
+    existingToken = ZERO_ADDR
+
+    // commonPool = totalSeedsAmount * initialRatio / (1 - initialRatio)
+    const totalSeedsAmount = tokens.holders.reduce(
+      (acc, [_, stake]) => acc + stake,
+      0
+    )
+    const initialRatio = issuance.initialRatio / 100
+    commonPool = (totalSeedsAmount * initialRatio) / (1 - initialRatio)
+
+    gardenTokenLiquidity = liquidity.tokenLiquidity
+    existingTokenLiquidity = '0'
```

Comment:
Here i used the same order that we have for the params passed to the contract function.
If it´s not confusing would like to keep as is, since we also have this order on the rest of the tx functions.